src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1778,6 +2050,10 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
                 Debug.Assert(!IsConditionalState);
                 _resultType = InferResultNullability(binary, leftType, rightType);
 
+                // For nested binary operators, this can be the only time they're visited due to explicit stack used in AbstractFlowPass.VisitBinaryOperator,
+                // so we need to set the flow-analysed type here.
+                _analyzedNullabilityMap[binary] = _resultType;
```

Comment:
>_analyzedNullabilityMap[binary] = _resultType [](start = 16, length = 45)

Consider adding a method to set the map entry, and replace explicit use of `_analyzedNullabilityMap`.

That method could also check we haven't set the same entry twice (we could clear the map when iterating through the method a second time), and we could skip the map altogether for callers that don't need it.

Reply:
I'll take an action item to do this in the future. Currently, there are definitely a few cases where we can see an item twice, because a specific scenario caused me to put an extra set in a more general location, as I figured that it would be better to double-get some items than not get one at all. We can assert that it doesn't _change_ after addition though.

---
In reply to: [260065585](https://github.com/dotnet/roslyn/pull/33591#discussion_r260065585) [](ancestors = 260065585)